src/main/java/com/google/devtools/build/lib/bazel/repository/downloader/SuperChecksum.java

```diff
@@ -0,0 +1,32 @@
+package com.google.devtools.build.lib.bazel.repository.downloader;
+
+import com.google.common.base.Optional;
+import com.google.common.hash.HashCode;
+import com.google.devtools.build.lib.bazel.repository.cache.RepositoryCache.KeyType;
+import com.google.devtools.build.lib.bazel.repository.downloader.Checksum;
+public class SuperChecksum {
```

Comment:
Please add documentation for this class. Right now, it appears to be a utility class that either stores null or a checksum inside it. I don't see the benefit of that versus a static method like `@Nullable Checksum maybeGetChecksum(KeyType keyType, String hash)`.

nit: put a newline before the class declaration.